scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -56,14 +56,19 @@
   [compute-cluster node-name->node node-name->pods]
   (let [compute-cluster-name (cc/compute-cluster-name compute-cluster)
         node-name->capacity (api/get-capacity node-name->node)
-        node-name->consumed (api/get-consumption node-name->pods)
+        ; Note that node-name->pods may include extra pods that are from other pools. So, we filter out only
+        ; nodes that have a capacity.
+        node-name->consumed (->> (api/get-consumption node-name->pods)
+                                 (filter #(node-name->capacity (first %)))
+                                 (into {}))
```

Comment:
Why does filtering to only nodes that have capacity fix the pods from other pools problem? I don't understand the connection. Is there a way to make the comment more clear? Also, should the comment say "filter in" instead of "filter out"?

Reply:
Can we write a unit test for this?

Reply:
Done. there should be no functionality change with this. The filter path is now tested with the nodeD change --- we have an off-node pod, which shouldn't (and doesn't) alter the offer generation.

Reply:
Did you see the first comment in this thread?

> Why does filtering to only nodes that have capacity fix the pods from other pools problem? I don't understand the connection. Is there a way to make the comment more clear? Also, should the comment say "filter in" instead of "filter out"?

Reply:
Comment elaborated.